app/controllers/logs_controller.rb

```diff
@@ -1,14 +1,46 @@
 class LogsController < ApplicationController
   def index
-    logs = UseCases::Administrator::GetAuthRequests.new(
+    if location_outside_organisation?
+      redirect_to(location_new_logs_search_path) && return
+    end
+
+    if valid_search_params.blank?
+      redirect_to(new_logs_search_path) && return
+    end
+
+    @location_name = Location.find_by(id: params[:location])&.address
+
+    @logs = get_auth_requests
+      .execute(ips: ip_params, username: params[:username])
+      .fetch(:results)
+  end
+
+private
+
+  def location_outside_organisation?
+    return unless params[:location].present?
+
+    location = Location.find(params[:location])
+    location.organisation != current_organisation
+  end
+
+  def valid_search_params
+    params.keys & %w[ip username location]
+  end
+
+  def get_auth_requests
+    UseCases::Administrator::GetAuthRequests.new(
       authentication_logs_gateway: Gateways::Sessions.new(
         ips: current_organisation.ips.map(&:address)
       )
-    ).execute(
-      username: params[:username],
-      ip: params[:ip]
     )
+  end
 
-    @logs = logs.fetch(:results)
+  def ip_params
+    if params[:location].present?
+      Location.find(params[:location]).ips.map(&:address)
+    else
+      params[:ip] ? [params[:ip]] : nil
+    end
```

Comment:
Still wonder if this above is the responsibility of the use case.
Happy to discuss but feel like some of its responsibilities have leaked out into this controller.

Reply:
Mm, it's a lot easier to sink some of this into the use case now it's cleaner, but there were some challenges around it that stopped me. 

I'll address the other things, look at this again, then let's chat.

Reply:
This was a good spot, I've rearranged this into something less painful looking now. Think the code that deals with "make a location into an ip list" now looks a lot cleaner.